docs/source/examples/classification_and_regression.rst

```diff
@@ -0,0 +1,302 @@
+    w, h = 218, 178
+    (x_L, y_L) = train_dataset[sample_number][1][1][0:2]
+    (x_R, y_R) = train_dataset[sample_number][1][1][2:4]
+    w_scale = image_size/w
+    h_scale = image_size/h
+    x_L, x_R = (x_L*h_scale), (x_R*h_scale)  #rescaling for the size of (224,224) and finaly to the range of [0,1]
+    y_L, y_R = (y_L*w_scale), (y_R*w_scale)
+    x_L, x_R = int(x_L), int(x_R)
+    y_L, y_R = int(y_L), int(y_R)
+    image_rgb	= cv2.drawMarker(image_rgb, (x_L,y_L), (0,255,0))
+    image_rgb	= cv2.drawMarker(image_rgb, (x_R,y_R), (0,255,0))
+    image_rgb = cv2.cvtColor(np.float32(image_rgb), cv2.COLOR_BGR2RGB)
+    image_rgb = np.clip(image_rgb ,0 , 1)
+    plt.imshow(image_rgb)
+    plt.axis('off')
+    plt.show()
+
+.. image:: /_static/img/classification_and_regression/dataset_sample.png
+
+Network
+=======
+
+Below, we define a new class, named 'ClassifierLocalizer, which accepts a pre-trained CNN and changes its last fully connected layer to be proper for the two task problem. The new fully connected layer contains 6 neurons, 2 for the classification task (male or female) and 4 for the localization task (x and y for the left and right eyes). Moreover, to put the location results on the same scale as the class scores, we apply the sigmoid function to the neurons assigned for the localization task.
+
+.. code-block:: python
+
+    class ClassifierLocalizer(nn.Module):
+        def __init__(self, model_name, num_classes=2):
+            super(ClassifierLocalizer, self).__init__()
+            self.num_classes = num_classes
+            
+            # create cnn model
+            model = getattr(models, model_name)(pretrained=True)
+            
+            # remove fc layers and add a new fc layer
+            num_features = model.fc.in_features
+            model.fc = nn.Linear(num_features, 6) # classifier + localizer
+            self.model = model
+        
+        def forward(self, x):
+            x = self.model(x)                    # extract features from CNN
+            scores = x[:, :self.num_classes]     # class scores
+            coords = x[:, self.num_classes:]     # coordinates
+            return [scores, torch.sigmoid(coords)]   # sigmoid output is in the range of [0, 1]
+
+Regarding the complexity of the problem, the number of the samples in the training dataset, and the similarity of the training dataset to the ImageNet dataset, we may decide to freeze some of the layers. In our current example, based on the mentioned factors, we freeze just the last fully connected layer.
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')
+
+    def freeze_weights(network):
+        for name, param in network.named_parameters():
+            if not name.startswith('fc.'):
+                param.requires_grad = False
+
+    freeze_weights(network)
+    print(network)
+
+.. code-block:: python
+
+    network = ClassifierLocalizer(model_name='resnet18')  # network without freezing any layer.
```

Comment:
Not sure why you freeze the layers if you create after with no frozen layers.

Reply:
I will remove that, I was just examining it as a hyper parameter